go/libraries/utils/async/action_executor.go

```diff
@@ -94,8 +101,10 @@ func (aq *ActionExecutor) Execute(val interface{}) {
 
 // WaitForEmpty waits until all the work that has been submitted before
 // the call to |WaitForEmpty| has completed. It returns any errors that
-// any actions may have encountered.
-func (aq *ActionExecutor) WaitForEmpty() error {
+// any actions may have encountered. If retainError is true, then the
+// returned error is not cleared, meaning repeat calls to WaitForEmpty
+// will result in the same error being returned.
+func (aq *ActionExecutor) WaitForEmpty(retainError bool) error {
```

Comment:
The added param and semantics strike me as kind of sloppy. AFAICT the only use for `true` could be avoided by making pkTableEditor.flush() return an error (which seems like a good idea anyway).